### PR TITLE
Ohm selection logic

### PR DESCRIPTION
**Pull request by Patharapol1573**
- attack state and move state logic for player and some preparation for enemy
- the branch use "" for specifying effect type such as "attack" or "move"

### CODE REVIEW

File: src/states/BattlePreparationState.py [hunk 1]
```diff
@@ -45,6 +45,9 @@ def Enter(self, params):
         self.enemy = Enemy("enemy")  
         self.enemy.deck = self.mockDeck()
 
+        self.player.fieldTile_index  = 2
+        self.enemy.fieldTile_index = 7
+
```
Comment (phukaokub) [inline src/states/BattlePreparationState.py:50]: @Patharapol1573 If these line are mocking data, please indicated with comment

File: src/states/BattleResolveState.py [hunk 2]
```diff
@@ -60,7 +61,7 @@ def update(self, dt, events):
             })
 
     def resolveCardEffect(self, effect, effectOwner):
-        if effect.type == EffectType.ATTACK:
+        if effect.type == "attack":
```
Comment (phukaokub) [inline src/states/BattleResolveState.py:64]: @Patharapol1573 Why change these from ENUM value to string value? The former handle with ENUM seem to be more reliable configurations

File: src/states/BattleResolveState.py [hunk 3]
```diff
@@ -71,7 +72,7 @@ def resolveCardEffect(self, effect, effectOwner):
                 'effect': effect,
                 'effectOwner': effectOwner
             })
-        elif effect.type == EffectType.MOVE:
+        elif effect.type == "move":
```
Comment (phukaokub) [inline src/states/BattleResolveState.py:75]: same as above

**Review by phukaokub — CHANGES_REQUESTED** [review]

**Review by rymTooo — APPROVED** [review]
Still kinda confusing on how the turn change or whose turn is it( or maybe I am just bad ). But everything seem to work fine though. There are some visual bug when entering the new turn and card on hand have 2 of them selected but as press arrow it disappear.

File: src/battleSystem/FieldTile.py [hunk 4]
```diff
@@ -6,6 +6,7 @@ def __init__(self, index, position):
         self.index = index  # Index in the row
         self.position = position  # (x, y) coordinates for rendering
         self.entity = None  # Store the entity occupying the fieldTile
+        self.rgb = (0,0,0)
```
Comment (krittintrs) [inline src/battleSystem/FieldTile.py:9]: i think using self.color might have better readability

File: src/states/BattleInitialState.py [hunk 5]
```diff
@@ -22,8 +22,8 @@ def Enter(self, params):
         self.currentTurnOwner = params['currentTurnOwner']  
 
         # Mock move entities
-        self.player.move_to(self.field[0], self.field)
-        self.enemy.move_to(self.field[8], self.field)
+        self.player.move_to(self.field[self.player.fieldTile_index], self.field)
```
Comment (krittintrs) [inline src/states/BattleInitialState.py:25]: is this for mock or for the actual movement?

File: src/states/SelectAttackState.py [hunk 6]
```diff
@@ -48,14 +120,25 @@ def update(self, dt, events):
...
-        
\ No newline at end of file
+            if fieldTile.index in set(self.avilableAttackTile):
+                fieldTile.color = (255,0,0)
+            else:
+                fieldTile.color = (0,0,0)
+            if self.selectAttackTile>=0:
+                if fieldTile.index == self.avilableAttackTile[self.selectAttackTile]:
+                    fieldTile.color = (255,0,255)
+                    fieldTile.solid = 0
```
Comment (krittintrs) [inline src/states/SelectAttackState.py:140]: will it be better if we define the color as constant variable for future change?

**Pull request merged**